source/tests/yats-test/pipeline_test.cpp

```diff
@@ -83,3 +83,31 @@ TEST(pipeline_test, add_multiple_listener)
     EXPECT_EQ(output1, 30);
     EXPECT_EQ(output2, 30);
 }
+
+TEST(pipeline_test, save_to_file)
+{
+    pipeline p;
+
+    struct add_task
+    {
+        slot<int, "sum"_id> run(slot<int, "summand_a"_id> a, slot<int, "summand_b"_id> b)
+        {
+            return a + b;
+        }
+    };
+
+    struct multiply_task
+    {
+        slot<int, "product"_id> run(slot<int, "factor_a"_id> a, slot<int, "factor_b"_id> b)
+        {
+            return a * b;
+        }
+    };
+
+    auto add_cfg = p.add<add_task>();
+    auto multiply_cfg = p.add<multiply_task>();
+
+    add_cfg->output<"sum"_id>() >> multiply_cfg->input<"factor_a"_id>();
+
+    p.save_to_file("graph.txt");
```

Comment:
this file should be deleted afterwards

Reply:
But then we cannot run DOT on it.
Maybe the test should be removed?